### PR TITLE
Storage/Datalake test coverage improvement

### PR DESCRIPTION
**Pull request by microzchang**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Comment by microzchang** [comment]
Mark changed test cases as LIVEONLY temporarily to wait for test records.

File: sdk/storage/azure-storage-files-datalake/test/ut/datalake_directory_client_test.cpp [hunk 1]
```diff
@@ -37,6 +37,34 @@ namespace Azure { namespace Storage { namespace Test {
...
+    auto clientOptions = InitStorageClientOptions<Files::DataLake::DataLakeClientOptions>();
+    {
+      auto directoryClient = Files::DataLake::DataLakeDirectoryClient::CreateFromConnectionString(
+          AdlsGen2ConnectionString(), m_fileSystemName, m_directoryName, clientOptions);
+      EXPECT_NO_THROW(directoryClient.GetProperties());
+    }
+    {
+      auto connectionStringParts = _internal::ParseConnectionString(AdlsGen2ConnectionString());
+      auto credential = std::make_shared<StorageSharedKeyCredential>(
+          connectionStringParts.AccountName, connectionStringParts.AccountKey);
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/test/ut/datalake_directory_client_test.cpp:51]:
```suggestion
      auto credential =_internal::ParseConnectionString(AdlsGen2ConnectionString());.KeyCredential;
```

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/test/ut/datalake_directory_client_test.cpp:51]: ditto for others

File: sdk/storage/azure-storage-files-datalake/test/ut/datalake_directory_client_test.cpp [hunk 2]
```diff
@@ -37,6 +37,34 @@ namespace Azure { namespace Storage { namespace Test {
     }
   } // namespace
 
+  TEST_F(DataLakeDirectoryClientTest, Constructors_LIVEONLY_)
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/test/ut/datalake_directory_client_test.cpp:40]: So you marked all of these as live-only and want to enable them in one shot with some script? Ok

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**